src/main/java/com/ibm/etcd/client/config/EtcdClusterConfig.java

```diff
@@ -103,11 +103,10 @@ private EtcdClient newClient() throws IOException, CertificateException {
         TlsMode ssl = tlsMode;
         if (ssl == TlsMode.AUTO || ssl == null) {
             String ep = endpointList.get(0);
-            if (ep.startsWith("http://")) ssl = TlsMode.PLAINTEXT;
-            else if (certificate != null || clientCertificate != null
-                    || ep.startsWith("https://")) {
-                ssl = TlsMode.TLS;
-            }
+            ssl = ep.startsWith("https://")
+                    || (!ep.startsWith("http://")
+                            && (certificate != null && clientCertificate != null))
```

Comment:
TLS client cert sometime is not required? Client may use other credentials to authenticate themselves.

Reply:
Yes client-auth is optional (and it wasn't possible to provide explicit certs for that until the last release).

But yes there's also etcd-level auth i.e. username/password which were always supported independent of this.

Reply:
Sorry @alienzach, I had misunderstood your question. You're right, the `&&` there should be `||`, great catch! Now pushed fix.